classes/class-dintero-checkout-subscription.php

```diff
@@ -635,6 +558,216 @@ public function show_payment_token( $order ) {
+		$profile_id      = $profile_id ?? self::get_subscription_profile_id();
+		$session_profile = self::get_subscription_session_profile( $profile_id );
+
+		// If the session profile is empty we cannot do anything.
+		if ( empty( $session_profile ) ) {
+			return $configuration;
+		}
+
+		return self::set_token_provider_from_profile( $session_profile, $configuration );
+	}
+
+	/**
+	 * Get the token provider as a string for the payment token request.
+	 *
+	 * @param string|null $profile_id The profile ID to use. If null, if not passed we will get it from the settings instead.
+	 *
+	 * @return string|null The token provider ID or null if none is found.
+	 */
+	public static function get_token_provider_string_from_profile( $profile_id = null ) {
+		$profile_id      = $profile_id ?? self::get_subscription_profile_id();
+		$session_profile = self::get_subscription_session_profile( $profile_id );
+
+		// If the session profile is empty we cannot do anything.
+		if ( empty( $session_profile ) ) {
+			return null;
+		}
+
+		$token_provider = self::get_token_provider_from_profile( $session_profile );
+		return ! empty( $token_provider ) ? "{$token_provider['id']}.{$token_provider['product_key']}" : null;
+	}
+
+	/**
+	 * Get the token provider string from an order if we have it stored.
+	 *
+	 * @param WC_Order $order The WooCommerce order.
+	 *
+	 * @return string|null The token provider ID or null if none is found.
+	 */
+	public static function get_token_provider_string_from_order( $order ) {
+		// Get the token provider from the metadata set on the order.
+		$token_provider = $order->get_meta( self::PAYMENT_PRODUCT_TYPE );
+
+		// If the token provider is still empty, get it from the session instead as a string.
+		if ( empty( $token_provider ) ) {
+			$token_provider = Dintero_Checkout_Subscription::get_token_provider_string_from_profile();
+		}
+
+		// Save the token provider to the order if we have it.
+		if ( ! empty( $token_provider ) ) {
+			$order->add_meta_data( self::PAYMENT_PRODUCT_TYPE, $token_provider );
+			$order->save_meta_data();
+		}
+
+		return $token_provider;
+	}
+
+	/**
+	 * Get the profile id to use for subscriptions.
+	 *
+	 * @return string The profile id.
+	 */
+	public static function get_subscription_profile_id() {
+		$settings = get_option( 'woocommerce_dintero_checkout_settings', array( 'profile_id' => '', 'subscription_profile_id' => '' ) );
+		return ! empty( $settings['subscription_profile_id'] ) ? $settings['subscription_profile_id'] : $settings['profile_id'];
+	}
+
+	/**
+	 * Get the session profile from Dintero for subscriptions.
+	 *
+	 * @param string $profile_id The profile ID to use.
+	 *
+	 * @return array|null The profile data or null if it could not be retrieved.
+	 */
+	private static function get_subscription_session_profile( $profile_id ) {
+		// If the profile ID is empty, we cannot do anything.
+		if ( empty( $profile_id ) ) {
+			return null;
+		}
```

Comment:
If the `profile_id` is empty, shouldn't we use the non-subscription profile ID instead?